e2e/common/traits/route_test.go

```diff
@@ -0,0 +1,382 @@
+		secret, err := createSecret(ns)
+		assert.Nil(t, err)
+
+		// they refer to the certificates create in the secret and are reused the different tests
+		refKey := secretName + "/" + corev1.TLSPrivateKeyKey
+		refCert := secretName + "/" + corev1.TLSCertKey
+
+		// =============================
+		// Insecure Route / No TLS
+		// =============================
+		t.Run("Route unsecure http works", func(t *testing.T) {
+			Expect(Kamel("run", "-n", ns, "files/PlatformHttpServer.java").Execute()).To(Succeed())
+			Eventually(IntegrationPodPhase(ns, integrationName), TestTimeoutMedium).Should(Equal(corev1.PodRunning))
+			route := Route(ns, integrationName)
+			Eventually(route, TestTimeoutMedium).ShouldNot(BeNil())
+			// must wait a little time after route is created, before doing an http request, 
+			// otherwise the route is unavailable and the http request will fail
+			time.Sleep(waitBeforeHttpRequest)
+			url := fmt.Sprintf("http://%s/hello?name=Simple", route().Spec.Host)
+			response, err := httpRequest(t, url, false)
+			assert.Nil(t, err)
+			assert.Equal(t, "Hello Simple", response)
+			Expect(Kamel("delete", "--all", "-n", ns).Execute()).Should(BeNil())
+		})
+
+		// =============================
+		// TLS Route Edge
+		// =============================
+
+		t.Run("Route Edge https works", func(t *testing.T) {
+			Expect(Kamel("run", "-n", ns, "files/PlatformHttpServer.java",
+				"-t", "route.tls-termination=edge",
+				"-t", "route.tls-certificate-secret=" + refCert,
+				"-t", "route.tls-key-secret=" + refKey,
+			).Execute()).To(Succeed())
+			Eventually(IntegrationPodPhase(ns, integrationName), TestTimeoutMedium).Should(Equal(corev1.PodRunning))
+			route := Route(ns, integrationName)
+			Eventually(route, TestTimeoutMedium).ShouldNot(BeNil())
+			// must wait a little time after route is created, before an http request, 
+			// otherwise the route is unavailable and the http request will fail
+			time.Sleep(waitBeforeHttpRequest)
+			url := fmt.Sprintf("https://%s/hello?name=TLS_Edge", route().Spec.Host)
+			response, err := httpRequest(t, url, true)
+			assert.Nil(t, err)
+			assert.Equal(t, "Hello TLS_Edge", response)
+			Expect(Kamel("delete", "--all", "-n", ns).Execute()).Should(BeNil())
+		})
+
+		// =============================
+		// TLS Route Passthrough
+		// =============================
+
+		t.Run("Route passthrough https works", func(t *testing.T) {
+			Expect(Kamel("run", "-n", ns, "files/PlatformHttpServer.java",
+				// the --resource mounts the certificates inside secret as files in the integration pod
+				"--resource", "secret:" + secretName + "@/etc/ssl/" + secretName,
+				// quarkus platform-http uses these two properties to setup the HTTP endpoint with TLS support
+				"-p", "quarkus.http.ssl.certificate.file=/etc/ssl/" + secretName + "/tls.crt",
+				"-p", "quarkus.http.ssl.certificate.key-file=/etc/ssl/" + secretName + "/tls.key",
+				"-t", "route.tls-termination=passthrough",
```

Comment:
Can't this be automated by the route trait? The user could be able to provide the secret name and the termination type.

Reply:
It just happen that using quarkus http requires these quarkus properties for [SSL setup](https://quarkus.io/guides/http-reference#ssl), whatever http client the user may choose it may require a different set of parameters to setup it. For example,   [Netty SSL](https://github.com/apache/camel-k/blob/main/examples/http/NettySecureServer.java#L18) requires mounting `-resource file:KeyStore.jks@/etc/ssl/keystore.jks `. With that said, it would be difficult to look at every SSL setup to automate in route trait.

Reply:
My point is more: could the trait set these properties automatically?

Reply:
Only if user is using quarkus http. If user uses netty, vertx or other http endpoint, they may require different SSL setup, for example the user may setup the `sslContextParameters` to read certificates from a custom path.

Reply:
Could the underlying HTTP endpoint used be inferred from the dependency set?

That would simplify a lot the configuration in that case, at least for Quarkus HTTP, as the mounting of the secret could be done automatically and the corresponding properties set accordingly.

Reply:
Can this be worked on another separated issue ?

Reply:
Sure, these tests are already a great move forwards.

Reply:
Create https://github.com/apache/camel-k/issues/2619 to track further investigation about this topic. Thanks for reviewing.